### PR TITLE
Deprecate level keyword for dataframe and series aggregations

### LINKED ISSUE #39983: Deprecate: `level` parameter for aggregations in DataFrame and Series
For `sum`, `prod`, `count`, etc. there is a parameter `level` which is internally rewritten into `groupby`. 

I propose that the `level` parameter for these aggregations be deprecated and instead recommend to users that they use the `groupby` syntax.

The functions that would be affected are:

* any
* all
* count
* sum
* prod
* max
* min
* mean
* median
* skew
* kurt
* sem
* var
* std
* mad

I believe this list is comprehensive.

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #39983
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

This would deprecate the level keyword if we want to move forward with this

### CODE REVIEW

File: doc/source/user_guide/advanced.rst [hunk 1]
```diff
@@ -492,6 +492,7 @@ Using the parameter ``level`` in the :meth:`~DataFrame.reindex` and
 values across a level. For instance:
 
 .. ipython:: python
+   :okwarning:
```
Comment (jreback) [inline doc/source/user_guide/advanced.rst:495]: would just remove this example

File: doc/source/user_guide/categorical.rst [hunk 2]
```diff
@@ -625,6 +625,7 @@ even if some categories are not present in the data:
 ``DataFrame`` methods like :meth:`DataFrame.sum` also show "unused" categories.
 
 .. ipython:: python
+    :okwarning:
```
Comment (jreback) [inline doc/source/user_guide/categorical.rst:628]: can you change this so its a regular group op

File: doc/source/user_guide/advanced.rst (hunk 1 above)
Comment (jreback) [reply doc/source/user_guide/advanced.rst:495]: *may* instead show a groupby example of this (but then i would put it in the groupby section), you can leave a pointer if you'd like here

File: doc/source/user_guide/groupby.rst [hunk 3]
```diff
@@ -325,6 +325,7 @@ directly. Additionally, the resulting index will be named according to the
 chosen level:
 
 .. ipython:: python
```
Comment (jreback) [inline doc/source/user_guide/groupby.rst:327]: point the example from advanced.rst here (and make this a proper groupby example)

File: pandas/core/frame.py [hunk 4]
```diff
@@ -9479,6 +9479,13 @@ def count(
         """
         axis = self._get_axis_number(axis)
         if level is not None:
+            warnings.warn(
+                "Using the level keyword in DataFrame and Series aggregations is "
+                "deprecated and will be removed in a future version. Use groupby "
```
Comment (jreback) [inline pandas/core/frame.py:9484]: can you add a pseudo example, eg.

```
df.sum(level=1) should use df.groupby(level=1).sum()
```

File: pandas/core/generic.py [hunk 5]
```diff
@@ -10260,6 +10260,13 @@ def _logical_func(
     ):
         nv.validate_logical_func((), kwargs, fname=name)
         if level is not None:
+            warnings.warn(
+                "Using the level keyword in DataFrame and Series aggregations is "
```
Comment (jreback) [inline pandas/core/generic.py:10264]: and same for all of these examples (change the example function to be applicable)

File: pandas/core/series.py [hunk 6]
```diff
@@ -1894,8 +1894,16 @@ def count(self, level=None):
...
         if level is None:
             return notna(self._values).sum()
-        elif not isinstance(self.index, MultiIndex):
-            raise ValueError("Series.count level is only valid with a MultiIndex")
+        else:
+            warnings.warn(
+                "Using the level keyword in DataFrame and Series aggregations is "
+                "deprecated and will be removed in a future version. Use groupby "
+                "instead.",
+                FutureWarning,
```
Comment (jreback) [inline pandas/core/series.py:1902]: is the test for Series.count both showing the deprecation AND raising

File: pandas/tests/frame/methods/test_count.py [hunk 7]
```diff
@@ -16,17 +16,22 @@ def test_count_multiindex(self, multiindex_dataframe_random_data):
         frame = frame.copy()
         frame.index.names = ["a", "b"]
 
-        result = frame.count(level="b")
-        expected = frame.count(level=1)
+        with tm.assert_produces_warning(FutureWarning):
```
Comment (jreback) [inline pandas/tests/frame/methods/test_count.py:19]: can you move these to a new test_count_with_level_deprecated

**Review by jreback — CHANGES_REQUESTED** [review]
wouldn't be averse to change some of the tests to use the groupby instead of the reduction method to remove / avoid the warnings (some, not all)

File: doc/source/user_guide/advanced.rst (hunk 1 above)
Comment (phofl) [reply doc/source/user_guide/advanced.rst:495]: If ok, I would keep it there, there are level examples above and below this snippet

Comment (phofl) [reply doc/source/user_guide/advanced.rst:495]: But changed to groupby

File: doc/source/user_guide/categorical.rst (hunk 2 above)
Comment (phofl) [reply doc/source/user_guide/categorical.rst:628]: Done

File: doc/source/user_guide/groupby.rst (hunk 3 above)
Comment (phofl) [reply doc/source/user_guide/groupby.rst:327]: Removed, since above is the groupby case

File: pandas/core/frame.py (hunk 4 above)
Comment (phofl) [reply pandas/core/frame.py:9484]: added

File: pandas/core/generic.py (hunk 5 above)
Comment (phofl) [reply pandas/core/generic.py:10264]: Done

File: pandas/core/series.py (hunk 6 above)
Comment (phofl) [reply pandas/core/series.py:1902]: Yes, should I change that?

File: pandas/tests/frame/methods/test_count.py (hunk 7 above)
Comment (phofl) [reply pandas/tests/frame/methods/test_count.py:19]: yep, done

**Comment by phofl** [comment]
> wouldn't be averse to change some of the tests to use the groupby instead of the reduction method to remove / avoid the warnings (some, not all)

Most of the tests are pretty basic and already covered in groupby, the more complicated ones are mostly checking if the aggregation with level equals a groupby, so don't know if this makes sense here?

**Comment by jreback** [comment]
> > wouldn't be averse to change some of the tests to use the groupby instead of the reduction method to remove / avoid the warnings (some, not all)
> 
> Most of the tests are pretty basic and already covered in groupby, the more complicated ones are mostly checking if the aggregation with level equals a groupby, so don't know if this makes sense here?

kk sounds good

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**